gen/definitions/generic/url_filtering_definition.yaml

```diff
@@ -0,0 +1,88 @@
+---
+name: URL Filtering Policy Definition
+rest_endpoint: /template/policy/definition/urlfiltering/
+has_version: true
+id_attribute: definitionId
+doc_category: Security Policies
+attributes:
+  - model_name: type
+    type: String
+    value: urlFiltering
+  - model_name: name
+    tf_name: name
+    type: String
+    mandatory: true
+    description: The name of the policy definition.
+    example: Example
+  - model_name: description
+    tf_name: description
+    type: String
+    mandatory: true
+    description: The description of the policy definition.
+    example: My description
+  - model_name: mode
+    tf_name: mode
+    type: String
+    enum_values: [security, unified]
+    default_value: 
+    description: The policy mode
+    example: security
+  - model_name: alerts
+    data_path: [definition]
+    tf_name: alerts
+    type: StringList
+    description: List of alerts options that will be exported as syslog messages
+    example: blacklist
+  - model_name: webCategories
+    data_path: [definition]
+    tf_name: web_categories
+    type: StringList
+    description: List of categories to block or allow
+    example: alcohol-and-tobacco
+  - model_name: webCategoriesAction
+    data_path: [definition]
+    tf_name: web_categories_action
+    type: String
+    enum_values: [block, allow]
+    description: whether the selected web categories should be blocked or allowed.
+    example: allow
+  - model_name: webReputation
+    data_path: [definition]
+    tf_name: web_reputation
+    type: String
+    enum_values: [high-risk, suspicious, moderate-risk, low-risk, trustworthy]
+    description: The web reputation of the policy definition
+    example: moderate-risk
+  - model_name: targetVpns
+    data_path: [definition]
+    tf_name: target_vpns
+    type: StringList
+    description: List of VPN IDs
+    example: 1
+  - model_name: ref
+    data_path: [definition, urlWhiteList]
+    tf_name: url_white_list_id
+    type: String
+    description: A white list ID
+    example: 84f10c9d-def7-45a3-8c64-6df26163c861
+    exclude_test: true
```

Comment:
Whenever we have a reference to another object we also add a version attribute (for example: https://github.com/CiscoDevNet/terraform-provider-sdwan/blob/6be026e2167b361e039aeba9d3ddbb6803fa216c/gen/definitions/generic/intrusion_prevention_policy_definition.yaml#L58), which is important to propagate changes to upstream objects. In this case it would look like this:
```
  - tf_name: allow_url_list_version
    tf_only: true
    type: Version
    description: Allow URL list version
    exclude_test: true
```